analyzers/src/SonarAnalyzer.CSharp/Rules/GenericReadonlyFieldPropertyAssignmentCodeFix.cs

```diff
@@ -38,21 +38,16 @@ public sealed class GenericReadonlyFieldPropertyAssignmentCodeFix : SonarCodeFix
     {
         internal const string TitleRemove = "Remove assignment";
         internal const string TitleAddClassConstraint = "Add reference type constraint";
-        public override ImmutableArray<string> FixableDiagnosticIds
-        {
-            get
-            {
-                return ImmutableArray.Create(GenericReadonlyFieldPropertyAssignment.DiagnosticId);
-            }
-        }
+        public override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(GenericReadonlyFieldPropertyAssignment.DiagnosticId);
 
-        private static readonly SyntaxAnnotation annotation = new SyntaxAnnotation(nameof(GenericReadonlyFieldPropertyAssignmentCodeFix));
+        private static readonly SyntaxAnnotation Annotation = new(nameof(GenericReadonlyFieldPropertyAssignmentCodeFix));
 
         protected override async Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext context)
         {
             var diagnostic = context.Diagnostics.First();
             var diagnosticSpan = diagnostic.Location.SourceSpan;
-            var memberAccess = (MemberAccessExpressionSyntax)root.FindNode(diagnosticSpan);
+            var memberAccess = (MemberAccessExpressionSyntax)root.FindNode(diagnosticSpan, getInnermostNodeForTie: true);
```

Comment:
This change was discovered by the C#10 test for the fixer. It was throwing an `InvalidCastException` before.